pkg/wasm/imagefetcher.go

```diff
@@ -61,14 +67,30 @@ func NewImageFetcher(ctx context.Context, opt ImageFetcherOption) *ImageFetcher
 	} else {
 		fetchOpts = append(fetchOpts, remote.WithAuth(&authn.Basic{Username: opt.Username}))
 	}
+
+	if opt.Insecure {
+		t := remote.DefaultTransport.Clone()
+		t.TLSClientConfig = &tls.Config{
+			InsecureSkipVerify: opt.Insecure, //nolint: gosec
+		}
+		fetchOpts = append(fetchOpts, remote.WithTransport(t))
+	}
+
 	return &ImageFetcher{
+		insecure:  opt.Insecure,
 		fetchOpts: append(fetchOpts, remote.WithContext(ctx)),
 	}
 }
 
 // Fetch is the entrypoint for fetching Wasm binary from Wasm Image Specification compatible images.
 func (o *ImageFetcher) Fetch(url, expManifestDigest string) ([]byte, error) {
-	ref, err := name.ParseReference(url)
+	nameOpts := make([]name.Option, 0, 1)
+	// support plain text repo, see more detail [here](https://github.com/google/go-containerregistry/blob/master/pkg/name/registry.go#L75)
+	if o.insecure {
+		nameOpts = append(nameOpts, name.Insecure)
```

Comment:
a bit confused - here we have plaintext and above we have `InsecureSkipVerify`. Do we support both TLS without verify _and_ plaintext? If so, how does the library decide which to use?

Reply:
without this, `ref` still use `https` protocol default.

https://github.com/google/go-containerregistry/blob/master/pkg/name/registry.go#L75
```golang
// Scheme returns https scheme for all the endpoints except localhost or when explicitly defined.
func (r Registry) Scheme() string {
	if r.insecure {
		return "http"
	}
	if r.isRFC1918() {
		return "http"
	}
	if strings.HasPrefix(r.Name(), "localhost:") {
		return "http"
	}
	if reLocal.MatchString(r.Name()) {
		return "http"
	}
	if reLoopback.MatchString(r.Name()) {
		return "http"
	}
	if reipv6Loopback.MatchString(r.Name()) {
		return "http"
	}
	return "https"
}
```

Reply:
So `if insecure use http`. But doesn't that mean if I have a registry at https://foo.bar that I want to use TLS with insecureSkipVerify, now it will use HTTP, not TLS?

Reply:
updated, ptal